sentence_transformers/SentenceTransformer.py

```diff
@@ -594,9 +632,6 @@ def tokenize(self, texts: Union[List[str], List[Dict], List[Tuple[str, str]]]):
         Tokenizes the texts
         """
         kwargs = {}
-        # HPU models reach optimal performance if the padding is not dynamic
-        if self.device.type == "hpu":
-            kwargs["padding"] = "max_length"
```

Comment:
Maybe it's best to keep the existing behaviour for now? Unless "max_length" is worse than `padding=True`, but I doubt that.

Reply:
No need for this now because we are proposing to use new additional padding to align the batch length into few power2 levels like [4, 8, 16, 32, 128 ..] still under the 'padding=True'  setting. It's kind of 'max_length' strategy used here to tradeoff between padding overhead and hpu alignment requirement.

Reply:
Makes total sense!